util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java

```diff
@@ -0,0 +1,106 @@
+package io.kubernetes.client.informer;
+
+import com.google.gson.reflect.TypeToken;
+import com.squareup.okhttp.Call;
+import io.kubernetes.client.*;
+import io.kubernetes.client.informer.impl.DefaultSharedIndexInformer;
+import io.kubernetes.client.util.CallGeneratorParams;
+import io.kubernetes.client.util.Watch;
+import io.kubernetes.client.util.common.Collections;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.function.Function;
+
+public class SharedInformerFactory {
+
+  private Map<Type, SharedIndexInformer> informers;
+
+  private Map<Type, Future> startedInformers;
+
+  private ExecutorService informerExecutor;
+
+  /** Constructor. */
+  public SharedInformerFactory() {
+    informers = new ConcurrentHashMap<>();
+    startedInformers = new HashMap<>();
+    informerExecutor = Executors.newCachedThreadPool();
```

Comment:
Probably want to add a constructor where the user can supply their own ThreadPool (for testing, or performance sensitive applications)

Reply:
added. also applied this comment to `SharedProcessor`.